src/ImageSharp/Common/Helpers/Numerics.cs

```diff
@@ -474,21 +474,8 @@ private static void ClampImpl<T>(Span<T> span, T min, T max)
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static void Premultiply(ref Vector4 source)
     {
-        float w = source.W;
-        source *= w;
-        source.W = w;
-    }
-
-    /// <summary>
-    /// Reverses the result of premultiplying a vector via <see cref="Premultiply(ref Vector4)"/>.
-    /// </summary>
-    /// <param name="source">The <see cref="Vector4"/> to premultiply</param>
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static void UnPremultiply(ref Vector4 source)
-    {
-        float w = source.W;
-        source /= w;
-        source.W = w;
+        Vector4 alpha = PermuteW(source);
+        source = WithW(source * alpha, alpha);
```

Comment:
You'll want to load `source` into a local here.

```suggestion
        Vector4 src = source;

        Vector4 alpha = PermuteW(src);
        source = WithW(src * alpha, alpha);
```

The reasoning is that since `source` is a `ref`, each access ends up coming from memory. However, we're using the value multiple times (twice given the implementation) and so accessing it twice is unnecessary.

Reply:
-- Since it's `AggressiveInlining`, its possible the JIT is already optimizing this almost equivalently. However, it might also make it more likely to spill the value since its "address taken" with multiple uses.